lsp/src/main/scala/org/ensime/lsp/rpc/RpcFormats.scala

```diff
@@ -0,0 +1,134 @@
+package org.ensime.lsp.rpc
+
+import spray.json._
+import shapeless._
+import messages._
+import JsonRpcMessages._
+import org.ensime.lsp.JsonUtils
+import shapeless.tag.@@
+
+import scala.util.Try
+
+private object RpcConversions extends DefaultJsonProtocol with FamilyFormats {
+
+  implicit override def eitherFormat[A: JsonFormat, B: JsonFormat]
+    : JsonFormat[Either[A, B]] = super.eitherFormat[A, B]
+
+  implicit val innerJsValue: JsonFormat[JsValue @@ JsInnerField] =
```

Comment:
I couldn't serialize simple `JsValue` (it was ambiguous implicit resolution), so added tag to it. Maybe there is some simpler solution?

Reply:
`JsonFormat` is invariant in the type parameter, so it's strange that this happens. Can you show a snippet that fails?

If it's actually problematic, you can try prioritizing the implicits in this object using inheritance (place lower priority implicits in another abstract class, and make `RpcConversions` extend that class)

Reply:
Without tags it throws following exception:
```
ambiguous implicit values:
[error]  both object JsValueFormat in trait AdditionalFormats of type org.ensime.lsp.rpc.RpcConversions.JsValueFormat.type
[error]  and method familyFormat in trait LowPriorityFamilyFormats of type [T, Repr](implicit gen: shapeless.LabelledGeneric.Aux[T,Repr], implicit sg: shapeless.Cached[shapeless.Strict[org.ensime.lsp.rpc.RpcConversions.WrappedRootJsonFormat[T,Repr]]])spray.json.RootJsonFormat[T]
[error]  match expected type spray.json.JsonFormat[spray.json.JsValue]
[error]   implicit val JsonRpcResponseSuccessMessageFormat
[error]                ^
```

Reply:
Ah. Hmm. I'm not sure what's the most elegant solution here. I can suggest:
1. Importing your formats at the call-site to prevent the `RootJsonFormat` from kicking in
2. Use your implicit explictly

Perhaps @fommil will have a nicer idea.